simulariumio/springsalad/springsalad_converter.py

```diff
@@ -96,11 +102,15 @@ def _parse_springsalad_data(
                 result.n_agents[time_index] += 1
                 result.unique_ids[time_index][agent_index] = int(cols[1])
                 raw_type_name = cols[3]
-                result.types[time_index].append(
-                    input_data.display_data[raw_type_name].name
-                    if raw_type_name in input_data.display_data
-                    else raw_type_name
-                )
+                if raw_type_name not in input_data.display_data:
```

Comment:
display_type_name = get_display_type_name_from_raw(raw_type_name, input_data.display_data)

Reply:
and put all the if-else logic inside that func

Reply:
I looked into doing this, and the issue with having a shared function is that for most of the converters, they have a specific way of setting up the `DisplayData` (including the `display_type_name`) given the `raw_type_name`, which is what's happening in the `if` clause here. I thought about trying to make an abstract method or passing a handler for that part and letting each converter implement it differently, but they mostly use different pieces of additional data to do it, so would need different arguments. It starts to seem like a lot of work to save a redundant `if` loop. But I can separate this into a function in each of the converters, it's a lot of logic to read directly in this main loop.

Reply:
@toloudis I made a function on `TrajectoryConverter` that a few of them can share (MCell, Smoldyn, SpringSaLaD), otherwise I just separated them into a new (unshared) function for readability, if it made sense.